include/grpc/grpc_crl_provider.h

```diff
@@ -0,0 +1,72 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPC_GRPC_CRL_PROVIDER_H
+#define GRPC_GRPC_CRL_PROVIDER_H
+
+#include <memory>
+#include <string>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace grpc_core {
+namespace experimental {
+
+// Opaque representation of a CRL
+class Crl {
+ public:
+  static absl::StatusOr<std::unique_ptr<Crl>> Parse(
+      absl::string_view crl_string);
+  virtual ~Crl() = default;
+
+ protected:
+  Crl() = default;
+
+ private:
+  std::string raw_crl_;
+};
+
+// Opaque representation of a Certificate
+class Cert {
```

Comment:
I don't understand the purpose of this class.  It's basically just a `string_view`, so why not just use that type directly instead of creating this wrapper?

Stepping back, it might be useful to consider what we expect a `CrlProvider` implementation to do with this.  Presumably, it needs to parse the cert to find the issuer, so that it knows which CRL to return.  How do we expect it to do that without reaching into OpenSSL internals?  As an alternative, consider instead having this class provide an abstract interface to the attributes of the parsed certificate, so that the parsing can happen on the gRPC side and we can expose the parsed data without using OpenSSL types.  What fields of the cert is the provider likely to need to look at?  Those are the methods we should provide here.

Reply:
Changed after offline discussion
In summary, we decided that passing `CertificateInfo` with information we need for CRL lookup is the best choice here. `CertificateInfo` is an extensible type. For now all it contains is the certificate's issuer.